app/models/mixins/cinder_manager_mixin.rb

```diff
@@ -0,0 +1,28 @@
+module CinderManagerMixin
+  extend ActiveSupport::Concern
+
+  included do
+    # TODO: how about many storage managers???
+    # Should use has_many :storage_managers,
+    has_one  :cinder_manager,
+             :foreign_key => :parent_ems_id,
+             :class_name  => "ManageIQ::Providers::StorageManager::CinderManager",
+             :autosave    => true,
+             :dependent   => :destroy
+
+    delegate :cloud_volumes,
+             :cloud_volume_snapshots,
+             :cloud_volume_backups,
+             :to        => :cinder_manager,
+             :allow_nil => true
+
+    private
+
+    def ensure_storage_managers
```

Comment:
@jerryk55 I don't think #2 is an option. At the top-level, it shouldn't know which storage managers are appropriate. For now, I say go with #1, but going forward, I think we may need to rethink this mixin thing.